proposals/dictionary-expressions.md

```diff
@@ -301,8 +217,8 @@ The new rules above represent a breaking change: For types that are a valid conv
 > **A create method will commonly use the name `CreateRange` in the dictionary domain.**
 >
 > For the create method:
->   - The method must have a single parameter of type System.ReadOnlySpan<E>, passed by value, and there is an identity conversion from E to the iteration type of the collection type.
->    - **The method has two parameters, where one is convertible to an `IEqualityComparer<TKey>` and the other follows the rules of the *single parameter* rule above. This method will be called if the collection expression's first element is an `expression_element` that is convertible to that parameter type.**
+>   - The method must have a single parameter of type System.ReadOnlySpan<E>, passed by value, and there is an identity conversion from E to the *iteration type* of the collection type. 
+>    - **The method can be called with two parameters, where one is a [*comparer*](Comparer-support) and the other follows the rules of the *single parameter* rule above. This method will be called if the collection expression's first element is an [*comparer*](Comparer-support) that is convertible to that parameter type.**
```

Comment:
note: i changed the wording subtly.  For constructors we just say it needs to be callable (meaning it could have optional arguments).  We can return this to just be exactly two parameters.  We can also be explicit if the comparer has to be the first parameter or not.  

This part is TBD when talking to runtime.  It will come down to if they want exactly one pattern used *everywhere*, or if they have a couple they expect.

Note: ToFrozenDictionary is of the form:

```c#
public static FrozenDictionary<TKey, TValue> ToFrozenDictionary<TKey, TValue>(this IEnumerable<KeyValuePair<TKey, TValue>> source, IEqualityComparer<TKey>? comparer = null)
```

While immutable dictionary looks like:

```
public static ImmutableDictionary<TKey, TValue> CreateRange<TKey, TValue>(IEqualityComparer<TKey>? keyComparer, IEnumerable<KeyValuePair<TKey, TValue>> items)
```

So, if we want to point at existing methods, we'd need to support these forms.  However, if we're mandating `ROS<KVP>`, then we'll need new overloads anyways, and we can have strict rules on the shape.

--

Finally, we may likelyy need new overloads as FrozenDict has overwrite semantics (which we want) while ImmutableDict does not (it throws).  

Tagging in @stephentoub .